src/main/java/seedu/address/logic/commands/AddDietCommand.java

```diff
@@ -16,43 +29,96 @@
+                                               + PREFIX_ALLERGY + "Egg "
+                                               + PREFIX_ALLERGY + "Crab "
+                                               + PREFIX_CULTURAL_REQUIREMENT + "Halal "
+                                               + PREFIX_PHYSICAL_DIFFICULTY + "Hands cannot move. ";
 
-    private final Person patientToEdit;
+    public static final String MESSAGE_SUCCESS = "Dietary requirements added for patient: %1$s";
+    public static final String MESSAGE_NO_SUCH_PATIENT = "No such patient exists.";
+    public static final String MESSAGE_MULTIPLE_PATIENTS = "Multiple such patients exist. "
+                                                           + "Please contact the system administrator.";
 
-    // TODO: add new class and include the attribute here
-    // private final Diet dietToAdd;
+    private final DietCollection dietsToAdd;
+    private final Nric patientNric;
 
     /**
      * Creates an AddDietCommand to add Diet to a patient.
-     * @param patient The patient to be edited.
+     * @param patientNric NRIC of the patient to be updated.
+     * @param dietCollection The collection of dietary requirements to be added to a patient..
      */
-    public AddDietCommand(Person patient) {
-        patientToEdit = patient;
+    public AddDietCommand(Nric patientNric, DietCollection dietCollection) {
+        this.patientNric = requireNonNull(patientNric);
+        this.dietsToAdd = requireNonNull(dietCollection);
     }
 
-    // TODO: finish this method
     @Override
     public CommandResult execute(Model model, CommandHistory history) throws CommandException {
-        return new CommandResult(MESSAGE_IN_PROGRESS);
+        requireNonNull(model);
+
+        ObservableList<Person> filteredByNric = model.getFilteredPersonList()
+                .filtered(p-> patientNric.equals(p.getNric()));
+
+        if (filteredByNric.size() < 1) {
+            throw new CommandException(MESSAGE_NO_SUCH_PATIENT);
+        }
+
+        if (filteredByNric.size() > 1) {
+            throw new CommandException(MESSAGE_MULTIPLE_PATIENTS);
+        }
+
+        Person patientToUpdate = filteredByNric.get(0);
+        Person updatedPatient = addDietsForPatient(patientToUpdate, dietsToAdd);
+
+        model.updatePerson(patientToUpdate, updatedPatient);
+        return new CommandResult(String.format(MESSAGE_SUCCESS, patientNric));
+    }
+
+    /**
+     * Update a patient with the new dietary requirement.
+     *
+     * @param patientToUpdate The patient to be updated.
+     * @param dietsToAdd The diet to be added to the patient.
+     * @return An updated patient with the appropriate diet added.
+     */
+    private static Person addDietsForPatient(Person patientToUpdate, DietCollection dietsToAdd) {
+        assert patientToUpdate != null;
+
+        DietCollection updatedDiet = patientToUpdate.getDietCollection().addMoreDiets(dietsToAdd);
```

Comment:
Consider making the getter return a defensive copy of the list to prevent alterations to the list from other sources.

Reply:
I will make another PR for this.